### DISCUSSION TITLE
.

### DISCUSSION THREAD

**Discussion by danielniccoli** [pydantic/pydantic#7644] [Feature Request]

**Comment by WayneYe** [comment]
+ 1️⃣ 

As of now, on pydantic 2.4.2:
```
API  |     from pydantic.alias_generators import to_camel
API  | ModuleNotFoundError: No module named 'pydantic.alias_generators'

```

**Reply by Viicos** [reply]
The import works fine on 2.4.2.

**Comment by xiaoFine** [comment]
I have the same problem and i found `serialization_alias` arg in `Field`. By this mean, i can init `Answer` with snake-case input and dump it in camel case
```python
class Answer(BaseModel):
    answer_id:int=Field(serialization_alias="answerId")
```
hope it helps

**Comment by Viicos** [comment]
You might be interested by `populate_by_name`:
https://docs.pydantic.dev/dev/api/config/#pydantic.config.ConfigDict.populate_by_name.

This will allow to use both camel/snake case when instantiating the object. But iirc it will also allow a mix of both (i.e. using camel case for a field, and snake case for another).

**Reply by Viicos** [reply]
Regarding
> I know that `model_dump_json()` accepts a parameter `by_alias=`, but why must it be that verbose?

You are making the assumption that `model_dump_json` is meant for the "outside" -- where in your case camel case is the convention used -- which isn't always the case.

> And why do I need `by_alias=True` in `model_dump_json()` to output it as `displayName` but `User('{"displayName": "Jon Doe"}')` is perfectly fine?

Although you probably meant `User.model_validate_json('{"displayName": "Jon Doe"}')`, this is a valid argument. I wonder if this could be made configurable (e.g. having the ability to define the default dumping behavior) without making things confusing. Unfortunately changing the default behavior of model_dump would be too much of a breaking change I think

**Reply by Viicos** [reply]
> Another way of thinking is: If you use a serialiser/deserialiser, you would expect that a string that is being deserialised into an object and immediately serialised into a string, remains the exact same string :)

yeah, makes sense
> I can't verify if that's what I meant, or if we're just having a communication problem.

This was purely about the pydantic call, `User('{"displayName": "Jon Doe"}')` is not supported by Pydantic, but instead you can pass the JSON string to `model_dump_json`.
